src/Core/Domain/Product/Command/AddProductCommand.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductType;
+
+/**
+ * Command for creating product with basic information
+ */
+class AddProductCommand
+{
+    /**
+     * @var string[]
+     */
+    private $localizedNames;
+
+    /**
+     * @var ProductType
+     */
+    private $type;
+
+    /**
+     * @param array $localizedNames
+     * @param int $type
+     */
+    public function __construct(
+        array $localizedNames,
+        int $type
+    ) {
+        $this->localizedNames = $localizedNames;
```

Comment:
I think we should also check that the array is not empty, but can be done in issue #19441 like the default language checking